src/district/server/async_db.clj

```diff
@@ -0,0 +1,34 @@
+(ns district.server.async-db)
+
+(defmacro with-async-resolver-tx [conn & body]
+  `(js/Promise.
+    (fn [resolve# reject#]
+      (.then (district.server.async-db/get-connection)
+             (fn [~conn]
+               (district.server.async-db/begin-tx ~conn)
+               (cljs.core.async/take! (district.shared.async-helpers/safe-go ~@body)
+                                      (fn [v-or-err#]
+
+                                        (if (cljs.core/instance? js/Error v-or-err#)
+                                          (do
+                                            (district.server.async-db/release-connection ~conn)
+                                            (reject# v-or-err#))
+
+                                          (do
+                                            (district.server.async-db/commit-tx ~conn)
```

Comment:
Shouldn't it release a connection after commiting the tx?

Reply:
Nice catch, fixed  also the rollback case